common/beerocks/bwl/dwpal/ap_wlan_hal_dwpal.cpp

```diff
@@ -1812,15 +1812,16 @@ bool ap_wlan_hal_dwpal::process_dwpal_event(char *buffer, int bufLen, const std:
         memset(msg_buff.get(), 0, sizeof(sACTION_APMANAGER_CLIENT_ASSOCIATED_NOTIFICATION));
         memset((char *)&msg->params.capabilities, 0, sizeof(msg->params.capabilities));
 
-        char VAP[SSID_MAX_SIZE]        = {0};
-        char MACAddress[MAC_ADDR_SIZE] = {0};
-        int supported_rates[16]        = {0};
-        int RRM_CAP[8]                 = {0};
-        int HT_MCS[16]                 = {0};
-        int16_t VHT_MCS[16]            = {0};
-        char ht_cap[8]                 = {0};
-        char vht_cap[16]               = {0};
-        size_t numOfValidArgs[20]      = {0};
+        char VAP[SSID_MAX_SIZE]                = {0};
+        char MACAddress[MAC_ADDR_SIZE]         = {0};
+        int supported_rates[16]                = {0};
+        int RRM_CAP[8]                         = {0};
+        int HT_MCS[16]                         = {0};
+        int16_t VHT_MCS[16]                    = {0};
+        char ht_cap[8]                         = {0};
+        char vht_cap[16]                       = {0};
+        size_t numOfValidArgs[20]              = {0};
+        char assoc_req[ASSOCIATION_FRAME_SIZE] = {0};
```

Comment:
Note that *here*, char is the correct type.

However, this size should be `2*ASSOCIATION_FRAME_SIZE+1` to fix the hex representation + the terminating \0.

Reply:
why 2 times ASSOCIATION_FRAME_SIZE?

Reply:
Because the 2048 was assumed to be the maximum of the *binary* frame, but it's represented as hex here. So you need two hex characters per byte + a terminating \0.